feature-review/automatic-subscriptions.md

```diff
@@ -0,0 +1,97 @@
+# Automatic Subscriptions - Feature Review
+
+Automatic subscriptions is a new way for clients to subscribe to streams whose properties meet some criteria, without having to call `room.subscribe()` for each of those streams.
+
+Clients will be able to use selectors, which will be based on stream properties, like `id`, `attributes`, etc. to decide which existing or future streams they want to subscribe in the room.
+
+## Changes in the Client API
+
+### Auto Subscriptions by Selector
+
+```js
+const selectors = {
+  '/id': '23',
+  '/attributes/group': '23',
+  '/attributes/kind': 'professor',
+  '/attributes/externalId': '10'
+};
+const options = {audio: true, video: false, forceTurn: true};
+room.autoSubscribe(selectors, options);
+
+// When stream meets one or more of the selectors and is finally subscribed.
+room.on('stream-subscribed', stream => {
+
+});
+
+// When a stream does not meet any selector anymore and is finally unsubscribed.
+room.on('stream-unsubscribed', stream => {
+
+});
+```
+
+### Update Stream attributes
+
+It already exists
+
+## Changes in the Server API
+
+### Update Stream attributes (2nd step)
+
+```js
+const roomId = '30121g51113e74fff3115502';
+const streamAttributes = [
+  {
+    selector: {
+      '/attributes/externalId': '10',
+    },
+    patch: [  // JSON Patch (http://jsonpatch.com/)
+      { op: 'remove', path: '/attributes/group'},
+      { op: 'add', path: '/attributes/kind', value: 'professor'},
+    ],
+  }
+];
+
+N.API.updateRoomStreamAttributes(roomId, streamAttributes, (resp) => {
+}, errorCallback);
+```
+
+## Architecture
+
+### Changes in ErizoController
+
+It handles requests to subscribe with selectors with the AutoSubscriptionHandler. AutoSubscriptionHandler will perform subscribe/unsubscribe actions based on the given selectors.
```

Comment:
I think I prefer `AutoSubscriptionManager` since it does not sound like a _handler_ to me.